test/export.service.test.js

```diff
@@ -98,3 +140,71 @@ describe('Check barebones bulk export logic (failure)', () => {
     await queue.close();
   });
 });
+
+describe('Check patient-level export logic (failure)', () => {
+  beforeEach(async () => {
+    await bulkStatusSetup();
+    await app.ready();
+  });
+  test('check 400 returned for invalid outputFormat', async () => {
+    await supertest(app.server)
+      .get('/Patient/$export?_outputFormat=invalid')
+      .expect(400)
+      .then(response => {
+        expect(JSON.parse(response.text).message).toEqual(
+          'The following output format is not supported for _outputFormat param for $export: invalid'
+        );
+      });
+  });
+
+  test('check 400 returned for invalid type (not a supported resource)', async () => {
+    await supertest(app.server)
+      .get('/Patient/$export?_type=invalid')
+      .expect(400)
+      .then(response => {
+        expect(JSON.parse(response.text).message).toEqual(
+          'The following resourceTypes are not supported for _type param for $export: invalid.'
+        );
+      });
+  });
+
+  test('check 400 returned when no resource types in _type param are included in patient compartment', async () => {
+    // Parameters is a supported resource that does not appear in the patient compartment
+    await supertest(app.server)
+      .get('/Patient/$export?_type=Parameters')
+      .expect(400)
+      .then(response => {
+        expect(JSON.parse(response.text).issue[0].details.text).toEqual(
+          'None of the provided resource types are permitted for Patient-level export.'
+        );
+      });
+  });
+
+  test('check 400 returned for unrecognized param', async () => {
+    await supertest(app.server)
+      .get('/Patient/$export?_unrecognizedparam=invalid')
+      .expect(400)
+      .then(response => {
+        expect(JSON.parse(response.text).message).toEqual(
+          'The following parameters are unrecognized by the server: _unrecognizedparam.'
+        );
+      });
+  });
+
+  test('DB is not populated when unrecognized param is in request', async () => {
+    const numEntries = await db.collection('bulkExportStatuses').countDocuments({});
+    await supertest(app.server)
+      .get('/Patient/$export?_unrecognizedparam=invalid')
+      .expect(400)
+      .then(async () => {
+        expect(await db.collection('bulkExportStatuses').countDocuments({})).toEqual(numEntries);
+      });
+  });
+
+  // Close export queue that is created when processing these tests
+  // TODO: investigate why queues are leaving open handles in this file
```

Comment:
Is this still happening? Wasn't for me.

Reply:
Now it is leaving an open handle for me after the successful tests, but I am able to remove these calls from the failure tests. I'm going to keep a `queue.close()` after the successful patient-level export tests finish so that we don't have the open handle issue.